crates/telemetry/src/state.rs

```diff
@@ -881,7 +1051,60 @@ use opentelemetry::sdk::metrics::aggregators::{histogram, last_value, sum};
 use opentelemetry::sdk::metrics::sdk_api::Descriptor;
 
 #[derive(Debug)]
-struct CustomAggregatorSelector;
+struct CustomAggregatorSelector {
+    tx_latency_submitted_range: Range<u64>,
+    tx_latency_submitted_buckets: u64,
+    tx_latency_confirmed_range: Range<u64>,
+    tx_latency_confirmed_buckets: u64,
+}
+
+impl CustomAggregatorSelector {
+    pub fn new(
+        tx_latency_submitted_range: Range<u64>,
+        tx_latency_submitted_buckets: u64,
+        tx_latency_confirmed_range: Range<u64>,
+        tx_latency_confirmed_buckets: u64,
+    ) -> Self {
+        Self {
+            tx_latency_submitted_range,
+            tx_latency_submitted_buckets,
+            tx_latency_confirmed_range,
+            tx_latency_confirmed_buckets,
+        }
+    }
+
+    pub fn get_submitted_range(&self) -> Vec<f64> {
+        let step = (self.tx_latency_submitted_range.end - self.tx_latency_submitted_range.start)
+            / self.tx_latency_submitted_buckets;
+        (0..=self.tx_latency_submitted_buckets)
+            .scan(self.tx_latency_submitted_range.start, |state, _| {
+                let current_value = *state;
+                if current_value >= self.tx_latency_submitted_range.end {
+                    return Some(self.tx_latency_submitted_range.end);
+                }
+                *state += step;
+                Some(current_value)
+            })
```

Comment:
GitHub Copilot came up with this instead, which I believe has the same semantics:

```suggestion
            (0..=self.tx_latency_submitted_buckets)
                .map(|i| self.tx_latency_submitted_range.start + i * step)
```